src/internal/pacherr/common.go

```diff
@@ -0,0 +1,62 @@
+package pacherr
+
+import (
+	"fmt"
+	"os"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+type ErrNotExist struct {
+	Collection string
+	ID         string
+}
+
+func NewNotExist(collection, id string) error {
+	return ErrNotExist{
+		Collection: collection,
+		ID:         id,
+	}
+}
+
+func (e ErrNotExist) Error() string {
+	return fmt.Sprintf("%s does not contain item: (%s)", e.Collection, e.ID)
+}
+
+func (e ErrNotExist) GRPCStatus() *status.Status {
+	return status.New(codes.NotFound, e.Error())
+}
+
+func IsNotExist(err error) bool {
+	target := ErrNotExist{}
+	return errors.As(err, &target) ||
+		os.IsNotExist(err)
```

Comment:
Why is `os.IsNotExist` included here?

Reply:
To make it more generally useful.  I don't think this would ever return a false positive.

Reply:
I would interpret `pacherr.IsNotExist` as telling me whether the type is `pacherr.ErrNotExist`, not whether it is either `pacherr.ErrNotExist` or `os.ErrNotExist`.

Reply:
The idea is that over time we migrate to this package for checking if an error is part of a certain class of errors.  If you want to know the type, you can use errors.Is, or cast from the interface.

Reply:
Okay, that might make sense. I think errors.Is is favored over these types of functions, but I guess it is fine for now.